lib/modules/datasource/maven/util.ts

```diff
@@ -139,6 +145,24 @@ export async function downloadS3Protocol(pkgUrl: URL): Promise<string | null> {
   return null;
 }
 
+export async function downloadArtifactRegistryProtocol(
```

Comment:
Besides the protocol replacement and `downloadHttpProtocol` call this code is pretty much identical to the one in the docker datasource **but** lacks exception handling that the other code includes:
https://github.com/renovatebot/renovate/blob/e32490787dbe448cb3c701e5664a2331599e851d/lib/modules/datasource/docker/common.ts#L113-L130

Instead of replicating it (without exception handling), I'd suggest to extract the existing from the docker datasource into a separate `getGoogleAuthToken()` (analogous to the existing `getECRAuthToken()`) method and put it into the common `util.ts`.